src/apps/chifra/pkg/types/types.go

```diff
@@ -105,32 +119,91 @@ type SimpleIndexAddressBelongs struct {
 }
 
 type SimpleLog struct {
-	Address          string   `json:"address"`
-	LogIndex         uint32   `json:"logIndex"`
-	BlockNumber      uint32   `json:"blockNumber,omitempty"`
-	TransactionIndex uint32   `json:"transactionIndex,omitempty"`
-	Timestamp        uint64   `json:"timestamp,omitempty"`
-	Topics           []string `json:"topics"`
-	Data             string   `json:"data,omitempty"`
-	CompressedLog    string   `json:"compressedLog,omitempty"`
+	Address          common.Address `json:"address"`
+	LogIndex         uint32         `json:"logIndex"`
+	BlockNumber      uint64         `json:"blockNumber,omitempty"`
+	TransactionIndex uint32         `json:"transactionIndex,omitempty"`
+	Timestamp        uint64         `json:"timestamp,omitempty"`
+	Topics           []common.Hash  `json:"topics"`
+	Data             string         `json:"data,omitempty"`
+	CompressedLog    string         `json:"compressedLog,omitempty"`
 }
 
 type SimpleReceipt struct {
 	BlockHash         common.Hash    `json:"blockHash"`
 	BlockNumber       uint64         `json:"blockNumber"`
-	ContractAddress   string         `json:"contractAddress,omitempty"`
+	ContractAddress   common.Address `json:"contractAddress,omitempty"`
 	CumulativeGasUsed string         `json:"cumulativeGasUsed"`
 	From              common.Address `json:"from"`
 	GasUsed           uint64         `json:"gasUsed"`
 	EffectiveGasPrice uint64         `json:"effectiveGasPrice"`
 	Logs              []SimpleLog    `json:"logs,omitempty"`
 	// LogsBloom         string         `json:"-"`
 	// Root              string         `json:"-"`
-	Status           *uint32     `json:"status"`
-	IsError          bool        `json:"isError,omitempty"`
-	To               string      `json:"to,omitempty"`
-	TransactionHash  common.Hash `json:"hash"`
-	TransactionIndex uint64      `json:"transactionIndex"`
+	Status           uint64         `json:"status"`
+	IsError          bool           `json:"isError,omitempty"`
+	To               common.Address `json:"to,omitempty"`
+	TransactionHash  common.Hash    `json:"hash"`
+	TransactionIndex uint64         `json:"transactionIndex"`
+	raw              *RawReceipt
+}
+
+func (r *SimpleReceipt) Raw() *RawReceipt {
+	return r.raw
+}
+
+func (r *SimpleReceipt) SetRaw(rawReceipt RawReceipt) {
+	r.raw = &rawReceipt
+}
+
+// To support custom format: just execute template on the output of Model
+func (r *SimpleReceipt) Model(showHidden bool, format string) Model {
```

Comment:
General question (I'm not sure I know what I'm talking about, so don't change anything based on this question). Since we have the `raw` data, can we use pointers to the fields of that raw data in the Simple structures? Would that allow us to avoid copying bytes?

Reply:
Yes, but: fields in SimpleTypes have different types than those in raw data. Everything in raw data is either a string or a slice of some type, whose fields are strings. This is how (Erigon at least) returns the data. So we need type casting to fill SimpleTypes with useful types (e.g. we probably want blockNumber to be a number instead of string)

Reply:
I see. I thing our simple types (other than BigNumbers) should be native types. So when the JSON is produced for the front end, it's native. If you remember, BigInt (i.e. wei) is different. Closing for now.